### DISCUSSION TITLE
Javascript Hook to Delete all records in a collection

### DISCUSSION THREAD

**Discussion by Mashiane** [pocketbase/pocketbase#4272] [Q&A]
Hi there

I need some help creating a hook that deletes all records in a collection in one go.

I have written this code (not tested) and am wondering if its correct.

`routerAdd("POST", "/deleteall/:tablename", (c) => {
    const tablename = c.pathParam("tablename")
    if (tablename.length > 0) {
        $app.dao().withoutHooks().runInTransaction((txDao) => {
            txDao.db().newQuery("DELETE " + tablename).execute()
        })
        return c.json(200, { "message": "Done " + tablename })
    }
    throw new BadRequestError("The tablename has not been specified!")
}, $apis.activityLogger($app))`

Please note I dont have any experience in Go but just trying this out.

Is there something I might be missing / doing incorrectly with this assumed code?

Thanks in advance.

**Comment by ganigeorgiev** [comment]
I don't have the full picture of your use case and the above seems vulnerable to SQL injections but I guess it is OK if you restrict it for admins only by adding the `$apis.requireAdminAuth()` after the `activityLogger` (see [JSVM - Builtin middlewares](https://pocketbase.io/docs/js-routing/#builtin-middlewares)).

I'm not sure if the above is just a rough excerpt of your actual code but note that you don't have to create explicitly dao without hooks if you are running raw SQL queries as there will be no events fired in this case (also you are maybe missing `FROM` in the delete statement).

**Comment by Mashiane** [comment]
Thanks for the clarity. I am using a collection for temporal storage and have to clear it first each time I have to redo my calculations, based on a user confirm event of clearing the content first. More like a working file.

Thing is the delete functionality supports only deletion by id per individual record and I need something that can just clear my collection. Thing is, one can view and report on the temporal collection data before they decide to clear data and recalculate.

I did think of deleting & recreating the collection based on the same schema it has but just could'nt figure the best approach and faster approach to handle this.

My code above is not actual but just a thought process and I appreciate for pointing the way forward.

**Comment by Mashiane** [comment]
There are variety of sql statements I thought about, e.g. delete where, using some other field(s) in the collection and even for a couple if ids at the same time e.g. delete from collection where id in (1,2,3,4,5) kinda thing. Of course I can always do a soft delete in some instances.

I would appreciate some heads up about such approaches. Thank you so much.

**Reply by tewshi** [reply]
answered here: https://github.com/pocketbase/pocketbase/issues/2238#issuecomment-2380877573

**Reply by Mashiane** [reply]
Thank you so much for the heads up. I did manage to get the routeAdd as Im still using v22.35 and have not migrated to any new version as yet. As my apps dont have many rules, just running .newQuery to execute direct sql commands on pocketbase works for me.
